contrib/checkbox-ce-oem/checkbox-provider-ce-oem/bin/wifi_test.py

```diff
@@ -0,0 +1,296 @@
+    if manager.key_mgmt != "none":
+        connect_cmd += " password {}".format(manager.ssid_pwd)
+    if manager.mode == "adhoc":
+        connect_cmd += " wifi.mode {}".format(manager.mode)
+    logging.info("Ping target Host first ...")
+    manager.run_command(ping_cmd(ip))
+    logging.info("Ping target Host %s successful...", ip)
+    logging.info("Attempting to connect DUT AP %s...", manager.ssid)
+    for i in range(1, 11):
+        logging.info(
+            "Attempting to connect DUT AP %s %d time...", manager.ssid, i
+        )
+        try:
+            manager.run_command(sshpass_cmd_gen(ip, user, pwd, connect_cmd))
+            logging.info("Connect successful!")
+            return True
+        except Exception:
+            logging.warning("Not able to found SSID {}".format(manager.ssid))
+        time.sleep(10)
+    logging.error("Not able to connect to DUT AP SSID {}".format(manager.ssid))
+    sys.exit(1)
+
+
+def ping_test(manager, ip, user, pwd):
+    try:
+        logging.info("Attempting to ping DUT...")
+        ping_result = manager.run_command(
+            sshpass_cmd_gen(ip, user, pwd, ping_cmd(manager.get_ip_addr()))
+        )
+        packet_loss = re.search(r"(\d+)% packet loss", ping_result).group(1)
+        logging.info("Packet loss: %s %%", packet_loss)
+        if packet_loss == "0":
+            logging.info("Ping DUT pass")
+            exit_code = 0
+        else:
+            logging.error("Ping DUT fail with %s %% packet loss!", packet_loss)
+            exit_code = 1
+    except Exception as e:
+        logging.error("An error occurred during ping_test: %s", str(e))
+        exit_code = 1
+    finally:
+        del_host_conn = "{} c delete {}".format(manager._command, manager.ssid)
+        try:
+            manager.run_command(sshpass_cmd_gen(ip, user, pwd, del_host_conn))
+            logging.info("Deleted host connection successfully.")
+        except Exception as e:
+            logging.error("Failed to delete host connection: %s", str(e))
+        sys.exit(exit_code)
+
+
+def main():
+    parser = argparse.ArgumentParser(description="WiFi test")
+
+    parser.add_argument(
+        "--type", required=False, default="wifi", help="Connection type."
+    )
+    parser.add_argument(
+        "--mode",
+        choices=["ap", "adhoc"],
+        required=False,
+        help="WiFi mode: ap or adhoc",
+    )
+    parser.add_argument(
+        "--interface", required=False, help="WiFi interface to use"
+    )
+    parser.add_argument("--band", required=False, help="WiFi band to use")
+    parser.add_argument(
+        "--channel", required=False, help="WiFi channel to use"
+    )
+    parser.add_argument(
+        "--keymgmt", required=False, help="Key management method"
+    )
+    parser.add_argument(
+        "--group", required=False, help="Group key management method"
+    )
+    parser.add_argument("--ssid", required=False, help="SSID for AP mode")
```

Comment:
I would suggest we initial the wireless SSID as a random, so we won't connect to wrong target if we perform this tests on multiple platform.
```suggestion
    parser.add_argument(
        "--ssid", 
        default= ''.join([random.choice(string.ascii_letters) for i in range(10)]),
        required=False, 
        help="SSID for AP mode")
```